### PR TITLE
fix: ClusterFuzzLite batch fuzzing linter fixes

### PR DESCRIPTION
**Pull request by AlisonB319**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/variables.go [hunk 1]
```diff
@@ -191,8 +202,14 @@ func getOutBoundCmd(nbc *datamodel.NodeBootstrappingConfiguration, cloudSpecConf
...
 	// so we need to use nc for the connectivity check.
-	clusterVersion, _ := semver.Make(cs.Properties.OrchestratorProfile.OrchestratorVersion)
-	minVersion, _ := semver.Make("1.18.0")
+	clusterVersion, err := semver.Make(cs.Properties.OrchestratorProfile.OrchestratorVersion)
+	if err != nil {
+		return err.Error()
+	}
+	minVersion, err := semver.Make("1.18.0")
+	if err != nil {
+		return err.Error()
```
Comment (alexeldeib) [inline pkg/agent/variables.go:211]: the return value here is really used during node provisioning, you can't return errors here

File: pkg/agent/variables.go [hunk 2]
```diff
@@ -171,10 +171,21 @@ func isVHD(profile *datamodel.AgentPoolProfile) string {
 }
 
 func getOutBoundCmd(nbc *datamodel.NodeBootstrappingConfiguration, cloudSpecConfig *datamodel.AzureEnvironmentSpecConfig) string {
+	if nbc.ContainerService == nil {
+		return "nbc.ContainerService is nil"
```
Comment (alexeldeib) [inline pkg/agent/variables.go:175]: hrm...we may need a simpler way to fix these, we can't return error strings in any of these cases as-is

File: pkg/agent/utils.go [hunk 3]
```diff
@@ -259,6 +259,9 @@ func IsKubernetesVersionGe(actualVersion, version string) bool {
 
 func getCustomDataFromJSON(jsonStr string) string {
 	var customDataObj map[string]string
+	jsonStr = strings.ReplaceAll(jsonStr, "\r", "\r\n")
```
Comment (alexeldeib) [inline pkg/agent/utils.go:262]: what's this one actually fixing?

Comment (AlisonB319) [reply pkg/agent/utils.go:262]: The errors were 
`panic: invalid character '\r' in string literal`
`panic: invalid character '\n' in string literal`
`panic: invalid character '\t' in string literal`

File: pkg/agent/variables.go (hunk 1 above)
Comment (AlisonB319) [reply pkg/agent/variables.go:211]: I see, the linter was basically upset that the err wasn't checked. Would it be better to return an empty string?

**Comment by alexeldeib** [comment]
I like the thinking but I might head a slightly different direction:
- for functions where fuzzing results in a panic because e.g. we don't do proper error checking, we need to adjust function signatures all the way up and down the stack to account for that. the goal is the root func cannot be misused.
- many funcs don't validate their input. we can fix that, but we need to avoid real errors in provisioning logic to satisfy the fuzzer. this again kind of points to strutural refactoring vs quick additions.

I think this is probably a longer/slower effort of identifying areas to refactor, to keep all the functionality and error cases the same but with cleaner structure

File: pkg/agent/variables.go (hunk 2 above)
Comment (AlisonB319) [reply pkg/agent/variables.go:175]: Same with the comment above, any object with the potential to be nil needed to be checked. Same suggestion of returning an empty string?

File: pkg/agent/utils.go (hunk 3 above)
Comment (AlisonB319) [reply pkg/agent/utils.go]: (I fixed `jsonStr = strings.ReplaceAll(jsonStr, "\r", "\r\n")` to `jsonStr = strings.ReplaceAll(jsonStr, "\r", "\n")` was trying to see if it just needed to be a different new line, but it wasn't happy)

**Comment by cameronmeissner** [comment]
linting is currently passing, closing this for now.

**Pull request closed without merging**